packages/embed-services/src/EmbedUrl.spec.ts

```
@@ -0,0 +1,158 @@
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import { EmbedUrl, ContentType } from './EmbedUrl'
+
+describe('EmbedUrl', () => {
+  const dashboardUrl = 'https://example.com/dashboards/123'
+  const dashboardEmbedUrl = 'https://example.com/embed/dashboards/123'
+  const exploreUrl = 'https://example.com/explore/456'
+  const lookUrl = 'https://example.com/embed/looks/789'
+  const invalidUrl = 'https://example.com'
+
+  describe('constructor', () => {
+    it('should set searchParams and path properties if present', () => {
+      const url = `${dashboardUrl}?param1=value1`
+      const targetUrl = new EmbedUrl(url)
+      expect(targetUrl.searchParams).toEqual({ param1: 'value1' })
+      expect(targetUrl.path).toBe('/dashboards/123')
+    })
+
+    it('should support a target url without search params', () => {
+      const targetUrl = new EmbedUrl(lookUrl)
+      expect(targetUrl.searchParams).toEqual({})
+      expect(targetUrl.path).toBe('/embed/looks/789')
+    })
+
+    it('should correctly determine isDashboard, isExplore, and isLook properties', () => {
+      const dashboardTargetUrl = new EmbedUrl(dashboardUrl)
+      const exploreTargetUrl = new EmbedUrl(exploreUrl)
+      const lookTargetUrl = new EmbedUrl(lookUrl)
+
+      expect(dashboardTargetUrl.isDashboard).toBe(true)
+      expect(dashboardTargetUrl.isExplore).toBe(false)
+      expect(dashboardTargetUrl.isLook).toBe(false)
+
+      expect(exploreTargetUrl.isDashboard).toBe(false)
+      expect(exploreTargetUrl.isExplore).toBe(true)
+      expect(exploreTargetUrl.isLook).toBe(false)
+
+      expect(lookTargetUrl.isDashboard).toBe(false)
+      expect(lookTargetUrl.isExplore).toBe(false)
+      expect(lookTargetUrl.isLook).toBe(true)
+    })
+
+    it('should throw if target url is invalid', () => {
+      expect(() => new EmbedUrl(invalidUrl)).toThrowError(
+        'Unknown content type'
+      )
```

Nice test pattern, but I'm wondering if we should throw an error for a utility class that tells us whether or not a given URL is an embed url

---

Any ideas? I could set `contentType` to `invalid` and not error

---

I like the idea of Invalid

---

Alright. I'll do that and make `EmbedUrl.embedUrl()` throw if called

---

Good idea